plugins/inputs/smart/smart.go

```diff
@@ -146,31 +160,152 @@ var (
+			Name:  "Temperature_Sensor_2",
+			Parse: parseTemperatureSensor,
+		},
+		"Temperature Sensor 3": {
+			Name:  "Temperature_Sensor_3",
+			Parse: parseTemperatureSensor,
+		},
+		"Temperature Sensor 4": {
+			Name:  "Temperature_Sensor_4",
+			Parse: parseTemperatureSensor,
+		},
+		"Temperature Sensor 5": {
+			Name:  "Temperature_Sensor_5",
+			Parse: parseTemperatureSensor,
+		},
+		"Temperature Sensor 6": {
+			Name:  "Temperature_Sensor_6",
+			Parse: parseTemperatureSensor,
+		},
+		"Temperature Sensor 7": {
+			Name:  "Temperature_Sensor_7",
+			Parse: parseTemperatureSensor,
+		},
+		"Temperature Sensor 8": {
+			Name:  "Temperature_Sensor_8",
+			Parse: parseTemperatureSensor,
+		},
+	}
+
+	// to obtain Intel specific metrics from nvme-cli
+	intelAttributes = map[string]struct {
+		ID    string
+		Name  string
+		Parse func(acc telegraf.Accumulator, fields map[string]interface{}, tags map[string]string, str string) error
+	}{
+		"program_fail_count": {
+			Name: "Program_Fail_Count",
+		},
+		"erase_fail_count": {
+			Name: "Erase_Fail_Count",
+		},
+		"end_to_end_error_detection_count": {
+			Name: "End_To_End_Error_Detection_Count",
+		},
+		"crc_error_count": {
+			Name: "Crc_Error_Count",
+		},
+		"retry_buffer_overflow_count": {
+			Name: "Retry_Buffer_Overflow_Count",
+		},
+		"wear_leveling": {
+			Name:  "Wear_Leveling",
+			Parse: parseWearLeveling,
+		},
+		"timed_workload_media_wear": {
+			Name:  "Timed_Workload_Media_Wear",
+			Parse: parseTimedWorkload,
+		},
+		"timed_workload_host_reads": {
+			Name:  "Timed_Workload_Host_Reads",
+			Parse: parseTimedWorkload,
+		},
+		"timed_workload_timer": {
+			Name: "Timed_Workload_Timer",
+			Parse: func(acc telegraf.Accumulator, fields map[string]interface{}, tags map[string]string, str string) error {
+				return parseCommaSeparatedIntWithAccumulator(acc, fields, tags, strings.TrimSuffix(str, " min"))
+			},
+		},
+		"thermal_throttle_status": {
+			Name:  "Thermal_Throttle_Status",
+			Parse: parseThermalThrottle,
+		},
+		"pll_lock_loss_count": {
+			Name: "Pll_Lock_Loss_Count",
+		},
+		"nand_bytes_written": {
+			Name:  "Nand_Bytes_Written",
+			Parse: parseBytesWritten,
+		},
+		"host_bytes_written": {
+			Name:  "Host_Bytes_Written",
+			Parse: parseBytesWritten,
+		},
 	}
 )
 
+type NVMeDevice struct {
+	name         string
+	vendorID     string
+	model        string
+	serialNumber string
+}
+
 type Smart struct {
-	Path       string
-	Nocheck    string
-	Attributes bool
-	Excludes   []string
-	Devices    []string
-	UseSudo    bool
-	Timeout    internal.Duration
+	PathSmartctl     string
+	PathNVMe         string
+	Nocheck          string
+	EnableExtensions []string
+	Attributes       bool
+	Excludes         []string
+	Devices          []string
+	UseSudo          bool
+	Timeout          internal.Duration
 }
 
 var sampleConfig = `
   ## Optionally specify the path to the smartctl executable
-  # path = "/usr/bin/smartctl"
+  # path_smartctl = "/usr/bin/smartctl"
```

Comment:
need to provide backwards-support for "path" if we're going to change the name

Reply:
It would be easier to understand readme and code if there is just `path` for `smartctl` executable. But it would be inconsistent with `path_nvme`.

@ssoroka what do you think?

Reply:
keep `path_smartctl`, but add an undocumented 
```go
Path string `toml:"path"`
```
to the app and if it's set, set PathSmartctl to that value.

Ideally all the fields settable from toml should have the toml tags defined.

Reply:
Kept `path` undocumented for backwards-support as proposed. Added toml tags where as you suggested.